### PR TITLE
clientv3: remove redundant retries in Lease, set FailFast=true

### PR DESCRIPTION
**Pull request by gyuho**
Separate out from https://github.com/coreos/etcd/pull/8710.
c.f. https://github.com/coreos/etcd/issues/8691.

### CODE REVIEW

File: clientv3/retry.go [hunk 1]
```diff
@@ -168,7 +168,7 @@ type retryLeaseClient struct {
 	readRetry retryRPCFunc
 }
 
-// RetryLeaseClient implements a LeaseClient that uses the client's FailFast retry policy.
+// RetryLeaseClient implements a LeaseClient.
```
Comment (xiang90) [inline clientv3/retry.go]: then why do we need the leaseClient at line167?

Comment (xiang90) [reply clientv3/retry.go]: can we rename the readRetry to repeatableRetry? leaseGrant is not really read. 

we will have a retrypolicy: repeatable defined in the future.

Comment (gyuho) [reply clientv3/retry.go]: It has to call `rlc.LeaseClient` where `LeaseClient` is `pb.LeaseClient`?

Comment (xiang90) [reply clientv3/retry.go]: i mean why it is embedded. it is that way so users can call to it directly. but now everything should call through the func we define on retryleaseclient.

Comment (xiang90) [reply clientv3/retry.go]: can we hide the raw client by not embedding it?

**Comment by xiang90** [comment]
lgtm

File: clientv3/retry.go (hunk 1 above)
Comment (gyuho) [reply clientv3/retry.go]: Right, since retryleaseclient implements the interface, no need to embed. Changed it unexported.

**Pull request merged**